internal/build/phase_test.go

```diff
@@ -70,66 +73,87 @@ func testPhase(t *testing.T, when spec.G, it spec.S) {
 		var err error
 		docker, err = client.NewClientWithOpts(client.FromEnv, client.WithVersion("1.38"))
 		h.AssertNil(t, err)
-		subject, err = CreateFakeLifecycle(filepath.Join("testdata", "fake-app"), docker, logger)
+		lifecycle, err = CreateFakeLifecycle(filepath.Join("testdata", "fake-app"), docker, logger, repoName)
 		h.AssertNil(t, err)
+		phaseFactory = build.NewDefaultPhaseFactory(lifecycle)
 	})
 
 	it.After(func() {
-		h.AssertNil(t, subject.Cleanup())
+		h.AssertNil(t, lifecycle.Cleanup())
 	})
 
 	when("Phase", func() {
 		when("#Run", func() {
 			it("runs the subject phase on the builder image", func() {
-				phase, err := subject.NewPhase("phase")
+				configProvider, err := build.NewPhaseConfigProvider(phaseName, lifecycle)
```

Comment:
If we were using a fake implementation of our interface here, we could assert that the relevant config provider methods were in fact called, which would become the test of [our factory implementation](https://github.com/buildpacks/pack/blob/b321dd4c4643c933eca842444c2871a39d33a4f0/internal/build/phase_factory.go#L13-L15)

Reply:
Discussed with @simonjjones - we can imagine future work to refactor `phase` and `phase.Run()` to be more unit testable, so that we don't have to spin up a real docker container in order to assert that it was configured correctly. However we believe that work could be done outside of this PR.